dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/RequestUtils.java

```diff
@@ -545,50 +555,73 @@ static private Hashtable parseQueryString(QueryString qs, String encoding) {
                 String key = null; //PM92940 Start
                 String value = null;
 
-                if (decodeParamViaReqEncoding && (!encoding_is_ShortEnglish)) {
+                try {
+                    if (decodeParamViaReqEncoding && (!encoding_is_ShortEnglish)) {
 
-                    // data, start, (end-start),encoding, string
-                    key = parse_decode_Parameter(qs.getKey(), encoding, "paramKey");
+                        // data, start, (end-start),encoding, string
+                        key = parse_decode_Parameter(qs.getKey(), encoding, "paramKey");
 
-                    if (!qs.hasEquals()) {
-                        value = key != null ? EMPTY_STRING : null;
-                    } else {
-                        value = parse_decode_Parameter(qs.getValue(), encoding, "paramValue");
-                    }
+                        if (!qs.hasEquals()) {
+                            value = key != null ? EMPTY_STRING : null;
+                        } else {
+                            value = parse_decode_Parameter(qs.getValue(), encoding, "paramValue");
+                        }
 
-                    if (ignoreInvalidQueryString && ((value == null) || (key == null))) {
-                        continue;
-                    }
+                        if (ignoreInvalidQueryString && ((value == null) || (key == null))) {
+                            //reason to have this log behind the ignore flag - app wants to continue operation with bad pairs
```

Comment:
All you're doing here is adding additional traces for service ? Just an FYI the `ignoreInvalidQueryString` is not in the meta type and publically exposed as proper configuration. If this is something we tell customers to set we should look at adding it into the official metatype for WebContainer.

Reply:
This flag is used less often in service.  It is mainly for any application that wants to collect all the data (i.e all valid query strings) without failing the request.  Other sensitive applications typically reject these type of requests.

We can expose it if you like; it is an old and unpopular property.

Reply:
Let's not expose it right now, if it comes up in service in the future I'd like to have this conversation again!